app/controllers/api/base_controller.rb

```diff
@@ -68,7 +68,7 @@ def current_user
   end
 
   def require_user!
-    if current_user && !current_user.disabled?
+    if current_user && !current_user.disabled? && current_user.confirmed?
```

Comment:
This condition was unneccessary before because unconfirmed users had no way to obtain an access token.